server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceReconciler.java

```diff
@@ -227,9 +227,15 @@ private void allocateUnassigned() {
                     logger.trace("No eligible node found to assign shard [{}] amongst [{}]", shard, desiredNodeIds);
                 }
 
-                final var allocationStatus = UnassignedInfo.AllocationStatus.fromDecision(
-                    isThrottled ? Decision.Type.THROTTLE : Decision.Type.NO
-                );
+                final UnassignedInfo.AllocationStatus allocationStatus;
+                if (isThrottled) {
+                    allocationStatus = UnassignedInfo.AllocationStatus.DECIDERS_THROTTLED;
+                } else if (desiredBalance.isBalanceComputed(shard.shardId()) == false) {
```

Comment:
Kind of a nit but I'm wondering why this check isn't first. I think this would override the throttled state. Can we be throttled without having the balance computed? That seems strange.

Reply:
Good point